### PR TITLE
 Template axsiymmetric 1D strain calculators

### PR DESCRIPTION
**Pull request by gambka**
Refs #21614.

Still need to add AD tests

### CODE REVIEW

**Comment by bwspenc** [comment]
What's here so far looks good. You might want to fix the spelling of "axisymmetric" in the PR and commit name.

File: modules/tensor_mechanics/include/materials/ComputeAxisymmetric1DFiniteStrain.h [hunk 1]
```diff
@@ -10,31 +10,42 @@
...
+#include "ADCompute1DFiniteStrain.h"
 #include "SubblockIndexProvider.h"
 
+// select the appropriate class based on the is_ad boolean parameter
+template <bool is_ad>
+using Generic1DFiniteStrain =
+    typename std::conditional<is_ad, ADCompute1DFiniteStrain, Compute1DFiniteStrain>::type;
+
+template <bool is_ad>
+using GenericComputeIncrementalStrainBase = typename std::
```
Comment (dschwen) [inline modules/tensor_mechanics/include/materials/ComputeAxisymmetric1DFiniteStrain.h:22]: This should not be necessary

File: modules/tensor_mechanics/include/materials/ComputeAxisymmetric1DFiniteStrain.h [hunk 2]
```diff
@@ -67,7 +78,24 @@ class ComputeAxisymmetric1DFiniteStrain : public Compute1DFiniteStrain
...
+  using Material::_current_elem;
+  using Material::_q_point;
+  using Material::_qp;
+  using Material::coupledScalarComponents;
+  using Material::coupledScalarValueOld;
+  using Material::coupledValueOld;
+  using Material::getBlockCoordSystem;
+  using Material::isCoupled;
+  using Material::isCoupledScalar;
+  using Material::isParamValid;
```
Comment (dschwen) [inline modules/tensor_mechanics/include/materials/ComputeAxisymmetric1DFiniteStrain.h:94]: `usingMaterialMembers` (or material base...)

File: modules/tensor_mechanics/src/materials/ComputeAxisymmetric1DFiniteStrain.C [hunk 3]
```diff
@@ -47,52 +53,62 @@ ComputeAxisymmetric1DFiniteStrain::ComputeAxisymmetric1DFiniteStrain(
...
     _scalar_out_of_plane_strain.resize(nscalar_strains);
     _scalar_out_of_plane_strain_old.resize(nscalar_strains);
+
     for (unsigned int i = 0; i < nscalar_strains; ++i)
     {
-      _scalar_out_of_plane_strain[i] = &coupledScalarValue("scalar_out_of_plane_strain", i);
+      if constexpr (is_ad)
+        _scalar_out_of_plane_strain[i] = &adCoupledScalarValue("scalar_out_of_plane_strain", i);
+      else
+        _scalar_out_of_plane_strain[i] = &coupledScalarValue("scalar_out_of_plane_strain", i);
```
Comment (dschwen) [inline modules/tensor_mechanics/src/materials/ComputeAxisymmetric1DFiniteStrain.C:62]: Maybe add a coupled generic scalar value method?

**Comment by bwspenc** [comment]
I just ran into a need for this today. Do we just need to add documentation for this?

**Comment by bwspenc** [comment]
Actually, seems like the documentation failure might have been unrelated to these changes. I guess the main missing thing is tests.

**Pull request closed without merging**